src/db_state.rs

```diff
@@ -35,28 +35,36 @@ impl SsTableHandle {
         SsTableHandle { id, info }
     }
 
+    // Compacted (non-WAL) SSTs are never empty. They are created by compaction or
```

Comment:
I took a brief look at `compactor_executor.rs` just to confirm this. It does, indeed, look like `execute_compaction` can't create empty SSTs since current_writer.close() is only called if there are records in it. Just noting this. 👍